tests/MongoDB.Driver.Core.Tests/Core/Connections/ConnectionInitializerTests.cs

```diff
@@ -39,6 +43,28 @@ public ConnectionInitializerTests()
             _subject = new ConnectionInitializer("test", new [] { new CompressorConfiguration(CompressorType.Zlib) });
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void CreateIsMaster_should_return_isMaster_with_speculativeAuthenticate(
+            [Values("default", "SCRAM-SHA-256", "SCRAM-SHA-1")] string authenticatorType,
+            [Values(false, true)] bool async)
+        {
+            var credentials = new UsernamePasswordCredential(
+                source: "Pathfinder", username: "Barclay", password: "Barclay-Alpha-1-7-Gamma");
+            var authenticator = CreateAuthenticator(authenticatorType, credentials);
+            var connectionSettings = new ConnectionSettings(new[] { authenticator });
+
+            var isMasterDocument = _subject.CreateInitialIsMasterCommand(new[] { authenticator });
+
+            isMasterDocument.Should().Contain("speculativeAuthenticate");
+            var speculativeAuthenticateDocument = isMasterDocument["speculativeAuthenticate"].AsBsonDocument;
+            speculativeAuthenticateDocument.Should().Contain("mechanism");
+            var expectedMechanism = new BsonString(
+                authenticatorType == "default" ? "SCRAM-SHA-256" : authenticatorType);
```

Comment:
Get a wider screen so you don't have to split lines unnecessarily... :)

Reply:
Sure... it just bothers me when lines are made too short.

It's just much better when every statement fits on one line.

We all have 1920 x 1080 screens now.